apps/condo/domains/organization/integrations/sbbol/sync/requestTransactions.js

```diff
@@ -0,0 +1,261 @@
+const dayjs = require('dayjs')
+const { get, isEmpty } = require('lodash')
+const { validate: uuidValidate } = require('uuid')
+
+const { getLogger } = require('@open-condo/keystone/logging')
+const { getSchemaCtx } = require('@open-condo/keystone/schema')
+
+const { BANK_INTEGRATION_IDS } = require('@condo/domains/banking/constants')
+const { BankAccount, BankTransaction, BankContractorAccount } = require('@condo/domains/banking/utils/serverSchema')
+const { RUSSIA_COUNTRY } = require('@condo/domains/common/constants/countries')
+const { ISO_CODES } = require('@condo/domains/common/constants/currencies')
+const { dvSenderFields, INVALID_DATE_RECEIVED_MESSAGE } = require('@condo/domains/organization/integrations/sbbol/constants')
+const { SBBOL_IMPORT_NAME } = require('@condo/domains/organization/integrations/sbbol/constants')
+const { initSbbolFintechApi } = require('@condo/domains/organization/integrations/sbbol/SbbolFintechApi')
+
+const { ERROR_PASSED_DATE_IN_THE_FUTURE } = require('../constants')
+
+
+const logger = getLogger('sbbol/SbbolSyncTransactions')
+const isTransactionsReceived = (response) => (get(response, 'error.cause', '') !== 'STATEMENT_RESPONSE_PROCESSING')
+// ---------------------------------
+const FAILED_STATUS = 'failed'
+const IN_PROGRESS_STATUS = 'inProgress'
+const COMPLETED_STATUS = 'completed'
+// ^^ the code needed in the prototype. In MVP, synchronization status will be stored in BankSyncTask
+/**
+ * Connects new BankTransaction records for BankAccount according to transaction data from SBBOL.
+ *  @param {String} userId
+ *  @param {BankAccount[]} bankAccounts
+ *  @param {keystoneContext} context
+ *  @param {String} statementDate
+ *  @param {String} organizationId
+ */
+async function _requestTransactions ({ userId, bankAccounts, context, statementDate, organizationId }) {
+    const fintechApi = await initSbbolFintechApi(userId)
+
+    if (!fintechApi) return
+
+    const transactions = []
+
+    for (const bankAccount of bankAccounts) {
+        // ---------------------------------
+        await BankAccount.update(context, bankAccount.id, {
```

Comment:
Discussed today to move it into BankIntegrationContext, that will reflect status and some meta-information, as like as in `BillingIntegrationOrganizationContext`.

Reply:
```
const bankIntegrationContext = await BankIntegrationContext.getOne(context, { id: bankAccount.integrationContext.id })
await BankIntegrationContext.update(context, bankIntegrationContext.id, {
    meta: {
        syncTransactionsTaskStatus: IN_PROGRESS_STATUS
    }
}
```